src/transforms/pipelines/config.rs

```diff
@@ -169,15 +145,11 @@ impl EventTypeConfig {
     ) -> crate::Result<Option<InnerTopology>> {
         let mut result = InnerTopology::default();
         let mut next_inputs = inputs.to_vec();
-        for pipe_name in self.names() {
-            let pipeline_name = name.join(&pipe_name);
-            let pipeline_config = self
-                .pipelines
-                .get_mut(&pipe_name)
-                .ok_or_else(|| format!("Unable to find pipeline with name {:?}", pipe_name))?;
+        for (pipeline_index, pipeline_config) in self.0.iter_mut().enumerate() {
+            let pipeline_name = name.join(pipeline_index);
             let topology = pipeline_config
                 .expand(&pipeline_name, &next_inputs)?
-                .ok_or_else(|| format!("Unable to expand pipeline named {:?}", pipe_name))?;
```

Comment:
I think using the pipeline name field here might make it easier for users to find.